cmake/public/mkl.cmake

```
@@ -6,7 +6,11 @@ endif()
 
 add_library(caffe2::mkl INTERFACE IMPORTED)
 target_include_directories(caffe2::mkl INTERFACE ${MKL_INCLUDE_DIR})
+if(MKL_VERSION VERSION_GREATER_EQUAL "2024.0.0")
+target_link_libraries(caffe2::mkl INTERFACE MKL::MKL)
+else()
 target_link_libraries(caffe2::mkl INTERFACE ${MKL_LIBRARIES})
+endif()
```

Why not use `MKL::MKL` everywhere? 
 Also, please make indent blocks inside `if` `else` blocks
```suggestion
if(MKL_VERSION VERSION_GREATER_EQUAL "2024.0.0")
    target_link_libraries(caffe2::mkl INTERFACE MKL::MKL)
else()
    target_link_libraries(caffe2::mkl INTERFACE ${MKL_LIBRARIES})
endif()
```